### PR TITLE
Reenable OpenMP

### PR DESCRIPTION
**Pull request by sfiligoi**
Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/.github/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by wasade** [comment]
@ebolyen, are there any concerns for QIIME 2 with scikit-bio utilizing OpenMP?

**Comment by sfiligoi** [comment]
@mr-c Can you check the handling of icc?

File: setup.py [hunk 1]
```diff
@@ -28,13 +28,29 @@
...
+
+if not gcc:
+    try:
+        if os.environ['CC'] == "clang":
+            # note, the conda provideed clang is not detected here
+            # and this is on purpose, as MacOS clang is very different
+            # than conda-provised one (which is llvm based)
+            clang = True
+        elif os.environ['CC'] == "icc":
+            icc = True
```
Comment (mr-c) [inline setup.py:50]:
```suggestion
        elif "icc" in os.environ['CC']:
            icc = True
```

File: setup.py [hunk 2]
```diff
@@ -28,13 +28,29 @@
 
 clang = False
 icc = False
+gcc = True
+
 try:
-    if os.environ['CC'] == "clang":
-        clang = True
+    if os.environ['CC'] == "gcc":
+        gcc = True
```
Comment (mr-c) [inline setup.py:35]:
```suggestion
    if "gcc" in os.environ['CC']:
        gcc = True
```



I would check for `gcc` in `CC`, etc.. Not only for exact matches

```
$ dpkg -S bin/gcc-[0-9]
gcc-11: /usr/bin/gcc-11
gcc-10: /usr/bin/gcc-10
gcc-13: /usr/bin/gcc-13
gcc-12: /usr/bin/gcc-12
$ dpkg -S bin/*-gcc-[0-9]
gcc-12-i686-linux-gnu: /usr/bin/i686-linux-gnu-gcc-12
gcc-13: /usr/bin/x86_64-linux-gnu-gcc-13
gcc-12: /usr/bin/x86_64-linux-gnu-gcc-12
gcc-11: /usr/bin/x86_64-linux-gnu-gcc-11
gcc-10: /usr/bin/x86_64-linux-gnu-gcc-10
gcc-9: /usr/bin/x86_64-linux-gnu-gcc-9
```

File: setup.py [hunk 3]
```diff
@@ -102,6 +115,16 @@
 elif platform.system() == 'Windows':
     ssw_extra_compile_args.extend(['-openmp:experimental'])
 
+stats_extra_compile_args = []+ssw_extra_compile_args
+stats_extra_link_args = []
+if platform.system() != 'Windows':
+    if icc or sysconfig.get_config_vars()['CC'] == 'icc':
+        stats_extra_compile_args.extend(['-qopenmp'])
+        stats_extra_link_args.extend(['-qopenmp'])
```
Comment (mr-c) [inline setup.py:123]:
```suggestion
    if icc or "icc" in sysconfig.get_config_vars()['CC']:
        stats_extra_compile_args.extend(['-qopenmp-simd'])
        stats_extra_link_args.extend(['-qopenmp-simd'])
```

File: setup.py [hunk 4]
```diff
@@ -102,6 +115,16 @@
...
+    elif not clang:
+        stats_extra_compile_args.extend(['-fopenmp'])
+        stats_extra_link_args.extend(['-fopenmp'])
```
Comment (mr-c) [inline setup.py:126]:
```suggestion
    else:
        stats_extra_compile_args.extend(['-fopenmp-simd'])
        stats_extra_link_args.extend(['-fopenmp-simd'])
```
(supported by both GCC 4.9+ and clang 6+)

**Review by mr-c — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (sfiligoi) [reply setup.py:123]: Nope, we need proper openmp, not just sims.

File: setup.py (hunk 4 above)
Comment (sfiligoi) [reply setup.py:126]: Nope, we need proper openmp, not just sims.

File: setup.py (hunk 2 above)
Comment (sfiligoi) [reply setup.py:35]: Indeed, I do not want to match on x86_64-linux-gnu-gcc-13.
Only plain gcc.
Will add a comment to clarify that.

File: setup.py (hunk 1 above)
Comment (sfiligoi) [reply setup.py:50]: Using just
`"icc" in` 
is too simplistic
It could match a path name that contains icc

Comment (sfiligoi) [reply setup.py]: Added better compiler detection.
Should do what the above was supposed to achieve.

**Review by wasade — APPROVED** [review]

**Pull request merged**